### PR TITLE
Order request

### PR DESCRIPTION
**Pull request by Deepakkumar-kaaylabs**

### CODE REVIEW

File: src/shared/types.ts [hunk 1]
```diff
@@ -61,3 +61,13 @@ export type LookupCardTypes ={
   onSelectCard: (cardKey: number) => void
 };
 
+export type TableSchemaTyeps ={
```
Comment (MJyotheesh) [inline src/shared/types.ts:64]: TableSchemaTypes

File: src/styles/order-request.scss
Comment (MJyotheesh) [inline src/styles/order-request.scss:1]: don't declare colors here

File: src/styles/generics/table.scss
Comment (MJyotheesh) [inline src/styles/generics/table.scss:1]: don't declare colors here

File: src/styles/generics/pagination.scss
Comment (MJyotheesh) [inline src/styles/generics/pagination.scss:1]: don't declare colors here

File: src/shared/icon-helper/back-arrow-icon.tsx [hunk 2]
```diff
@@ -1,14 +1,68 @@
...
+      3.608H88.413v1.587H76.85l3.43,3.608-1.069,1.124-5.25-5.525,5.25-5.525Z"
+        transform="translate(-73.962 -177.616)"
+        fill="#1d1d1d"
+      />
     </svg>
-
   );
 };
+
+export const BackArrowIcon1 = (): JSX.Element =>
```
Comment (MJyotheesh) [inline src/shared/icon-helper/back-arrow-icon.tsx:32]: give proper name

File: src/shared/types.ts [hunk 3]
```diff
@@ -61,3 +61,13 @@ export type LookupCardTypes ={
...
+  label: string,
+  value: string,
+  renderer?: string
+}
+export type responseType = Promise<Response>
+export type apiReturnType<arg> = (arg1?: arg) => responseType
+export type apiReturnType1<arg1, arg2> = (arg1: arg1, arg2: arg2) => responseType
+export type apiReturnType2<arg1, arg2, arg3> = (arg1: arg1, arg2: arg2, arg3: arg3) => responseType
+
```
Comment (MJyotheesh) [inline src/shared/types.ts:73]: start type name with capital letter like -> ResponseType

File: src/shared/api/http-helper.ts
Comment (MJyotheesh) [inline src/shared/api/http-helper.ts:1]: resolve lint issues

File: src/shared/api/http.ts
Comment (MJyotheesh) [inline src/shared/api/http.ts:1]: resolve lint issues

File: src/pages/routing.tsx [hunk 4]
```diff
@@ -20,11 +20,11 @@ import {
 import ForgotPassword from "./forgot-password";
 
 const isUserLoggedIn = (): boolean => Boolean(getSessionStorage("admin"));
-const navigate = (path: string): any => {
+const navigate = (path: string): JSX.Element => {
   return <Navigate replace to={path} />;
 };
 
-const Routing = (props: { notification: any }): JSX.Element => {
+const Routing = (props: { notification: unknown }): JSX.Element => {
```
Comment (MJyotheesh) [inline src/pages/routing.tsx:27]: notification won't be unknown

File: src/pages/routing.tsx
Comment (MJyotheesh) [inline src/pages/routing.tsx:1]: resolve lint issues

File: src/pages/orders/order-request.tsx [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+          <div className="col-md-1 col-12 p-0" onClick={()=>{
+            navigate("/");
+          }}>
+            <BackArrowIcon1 />
+          </div>
+        }
+      />
+      <div className="table-grid">
+        <Table
+          colums={tableSchema}
```
Comment (MJyotheesh) [inline src/pages/orders/order-request.tsx:61]: check spelling

File: src/pages/login/index.tsx
Comment (MJyotheesh) [inline src/pages/login/index.tsx:1]: resolve lint issues

File: src/components/wrapper.tsx [hunk 6]
```diff
@@ -7,7 +7,7 @@ type ComponentProps = {
 }
 type WrapperProps = {
   Component: React.FC<ComponentProps|any>;
-  notification?: () => void;
+  notification?: unknown;
```
Comment (MJyotheesh) [inline src/components/wrapper.tsx:10]: it won't be unknown

File: src/components/generics/title-search-component.tsx
Comment (MJyotheesh) [inline src/components/generics/title-search-component.tsx:1]: resolve lint issues

File: src/components/generics/table.tsx [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+import Pagination from "./pagination";
+import {
+  type ObjectType,
+  type TableSchemaTyeps,
+  type responseType,
+} from "../../shared/types";
+import Spinner from "./spinner";
+import { NoDataIcon } from "shared/icon-helper";
+
+type Itable = {
```
Comment (MJyotheesh) [inline src/components/generics/table.tsx:11]: give proper type name

File: src/components/generics/table.tsx
Comment (MJyotheesh) [inline src/components/generics/table.tsx:1]: resolve lint issues don't disable lint

File: src/components/generics/pagination.tsx [hunk 8]
```diff
@@ -0,0 +1,86 @@
+import { type FC } from "react";
+type TPagination = {
```
Comment (MJyotheesh) [inline src/components/generics/pagination.tsx:2]: give PaginationType

File: src/shared/api/http-helper.ts
Comment (Deepakkumar-kaaylabs) [reply src/shared/api/http-helper.ts:1]: "strict-boolean-expressions"   can't able to solved, Can you please suggest a way for that

File: src/shared/api/http-helper.ts [hunk 9]
```diff
@@ -26,25 +27,25 @@ const createQuery = (obj: any) => {
...
   }
-  return Promise.reject(result);
+  return await Promise.reject(result);
 };
 
-const validateSuccess = (result: any) => {
-  const content = result.headers && result.headers.get("content-disposition");
+const validateSuccess = async (result: Response): responseType => {
+  const content = result.headers?.get("content-disposition");
   if (content) {
```
Comment (MJyotheesh) [inline src/shared/api/http-helper.ts:39]: content != null or undefined

Comment (MJyotheesh) [inline src/shared/api/http-helper.ts:39]: content != null

File: src/shared/api/http-helper.ts [hunk 10]
```diff
@@ -26,25 +27,25 @@ const createQuery = (obj: any) => {
...
-    return result.json();
+    return await result.json();
   }
-  return Promise.reject(result);
+  return await Promise.reject(result);
 };
 
-const validateSuccess = (result: any) => {
-  const content = result.headers && result.headers.get("content-disposition");
+const validateSuccess = async (result: Response): responseType => {
```
Comment (MJyotheesh) [inline src/shared/api/http-helper.ts:37]: (result: { success: boolean } & Response): ResponseType

File: src/shared/api/http-helper.ts
Comment (MJyotheesh) [reply src/shared/api/http-helper.ts:1]: Check below

File: src/components/generics/title-search-component.tsx
Comment (Deepakkumar-kaaylabs) [reply src/components/generics/title-search-component.tsx:1]: backIcon in the type of ReactNode so we can't able to resolve strict boolean expression

File: src/components/generics/button.tsx [hunk 11]
```diff
@@ -33,19 +33,19 @@ const ButtonComponent = ({
       style={{ ...style }}
       {...props}
     >
-      {prefix && (
+      {typeof prefix !== "undefined" && (
```
Comment (MJyotheesh) [inline src/components/generics/button.tsx:36]: prefix !== undefined

File: src/components/generics/button.tsx [hunk 12]
```diff
@@ -33,19 +33,19 @@ const ButtonComponent = ({
...
         <span className="btn" style={prefixStyle} onClick={onPrefixClick}>
           {prefix}
         </span>
       )}
-      {loading ? (
+      {typeof loading !== "undefined" ? (
```
Comment (MJyotheesh) [inline src/components/generics/button.tsx:41]: !== undefined

File: src/components/generics/button.tsx [hunk 13]
```diff
@@ -33,19 +33,19 @@ const ButtonComponent = ({
...
         <div className="spinner-border" role="status">
-        <span className="visually-hidden">Loading...</span>
-      </div>
+          <span className="visually-hidden">Loading...</span>
+        </div>
       ) : (
         <span className="icon-label ">{label}</span>
       )}
-      {suffix && (
+      {typeof suffix !== "undefined" && (
```
Comment (MJyotheesh) [inline src/components/generics/button.tsx:48]: !== undefined

File: src/components/wrapper.tsx [hunk 14]
```diff
@@ -1,14 +1,15 @@
...
 type ComponentProps = {
   notification?: () => void;
-}
+};
 type WrapperProps = {
-  Component: React.FC<ComponentProps|any>;
-  notification?: unknown;
-}
+  Component: React.FC<ComponentProps | any>;
+  notification?: NotificationTye;
```
Comment (MJyotheesh) [inline src/components/wrapper.tsx:11]: NotificationType

File: src/pages/order-request/index.tsx [hunk 15]
```diff
@@ -0,0 +1,74 @@
...
+        <NavLink className="order-request-id" to={"/home"}>
+          {value.orderRequestUid}
+        </NavLink>
+      );
+    },
+    viewDetails: (value: ObjectType): ReactNode => {
+      return (
+        <ButtonComponent
+          onClick={() => {
+            // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
```
Comment (MJyotheesh) [inline src/pages/order-request/index.tsx:26]: resolve this

File: src/pages/order-request/index.tsx
Comment (MJyotheesh) [inline src/pages/order-request/index.tsx:1]: no need separate folder for order requests move into orders folder itself

**Review by MJyotheesh — CHANGES_REQUESTED** [review]
Comments

File: src/pages/order-request/index.tsx
Comment (Deepakkumar-kaaylabs) [reply src/pages/order-request/index.tsx:1]: We have a grid and details page in order-request. so we maintain it sepratly

Comment (MJyotheesh) [reply src/pages/order-request/index.tsx:1]: move it inside orders because we don't have a individual route for order request page

**Review by MJyotheesh — APPROVED** [review]

**Pull request merged**